std/algorithm/mutation.d

```diff
@@ -1695,6 +1696,59 @@ enum SwapStrategy
     assert(arr == [10, 9, 8, 4, 5, 6, 7, 3, 2, 1]);
 }
 
+/**
+Checks whether a type is either an integral static array (e.g. int[2])
+or a tuple of two integral types
+*/
+private template isIntegralPair(T)
+{
+    import std.traits : isStaticArray;
+    static if (__traits(compiles, T.length > 0))
+    {
+        static if (T.length == 2)
+        {
+            static if (isStaticArray!T)
+                enum isIntegralPair = isIntegral!(typeof(T.init[0]));
+            else
+                enum isIntegralPair = T.length == 2 &&
+                            isIntegral!(typeof(T[0])) &&
+                            isIntegral!(typeof(T[1]));
+        }
+        else
+        {
+            enum isIntegralPair = false;
+        }
+    }
+    else
+    {
+        enum isIntegralPair = false;
+    }
+}
```

Comment:
On the bottom of this statement you use single-statement curly brace block for static else. But at line 1713 you newline after it without curly braces, a style considered error-prone by some. Should you change these to be consistent with each other? On the other hand, this might be an over-pedantic suggestion.

Reply:
The current convention is that curly braces only need to be used for more than one line, i.e. when they are required.
I don't mind changing but to my knowledge always requiring curly braces isn't part of the DStyle (as of now).

Reply:
And please don't make it part of the style. That would be incredibly annoying. Personally, I always use them when the condition is longer than one line, since it gets hard to read otherwise, but other than that, I never use braces when their contents are just one line long, and many other contributors to Phobos don't use the pointless braces either. They just add unnecessary vertical space to the file.

Reply:
I did not mean that lines like that should have the curly braces always, but that they should (maybe) be consistent with each other in the same function. Meaning, either use or don't use curly braces for if or else clause at one line and one statement at another line, but don't do both in the same function.